### PR TITLE
Add educational Pluto example

### PR DESCRIPTION
**Pull request by beorostica**
This pull request is intended to resolve issue #242

### CODE REVIEW

**Comment by cncastillo** [comment]
Change name of folder containing the example to `7.educational_pluto_notebook`

**Comment by beorostica** [comment]
> educational_pluto_notebook

Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/ba7b836a443d33d4759e2a7f727cfeac37e281b3

**Comment by beorostica** [comment]
> The solution looks like this:
> 
> image

Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/ba7b836a443d33d4759e2a7f727cfeac37e281b3

**Comment by beorostica** [comment]
> Change name of folder containing the example to `7.educational_pluto_notebook`

Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/ba7b836a443d33d4759e2a7f727cfeac37e281b3

File: docs/src/pluto-simulation.md [hunk 1]
```diff
@@ -7,5 +7,8 @@ This example was used in the workshop [MRI: Processing your own data](https://gi
 The participants used **KomaMRI** within a **Pluto** notebook to simulate some basic 1D sequences:  Free Induction Decay (FID), Gradient Echo (GE), and Spin Echo (SE). Try to solve the problems yourself! 😼
 
 ```@raw html
-<iframe type="text/html" src="../../../examples/pluto/simulation-example-empty.html" style="height:100vh;width:100%;"></iframe>
-
+<iframe type="text/html" src="../../../examples/educational_pluto_notebook/simulation-example-empty.html" style="height:100vh;width:100%;"></iframe>
+<details><summary>View Solution</summary>
+<iframe type="text/html" src="../../../examples/educational_pluto_notebook/simulation-example.html" style="height:100vh;width:100%;"></iframe>
+</details>
```
Comment (cncastillo) [inline docs/src/pluto-simulation.md:13]: Folder name missing number `7.educational_pluto_notebook`

File: examples/educational_pluto_notebook/simulation-example.html
Comment (cncastillo) [inline examples/educational_pluto_notebook/simulation-example.html:1]: Did you update the .jl as well? Are you using `PlutoUILocalResources` or `PlutoUI.Resource`?

File: docs/src/pluto-simulation.md (hunk 1 above)
Comment (beorostica) [reply docs/src/pluto-simulation.md]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/bc6b02f796cffa9a79a465bef5007a56ca538591

File: examples/educational_pluto_notebook/simulation-example.html
Comment (beorostica) [reply examples/educational_pluto_notebook/simulation-example.html:1]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/bc6b02f796cffa9a79a465bef5007a56ca538591

I updated the .jl too. I'm using `PlutoUI.Resource`

File: examples/6.educational_pluto_notebook/simulation-example.jl [hunk 2]
```diff
@@ -4,16 +4,6 @@
...
-# This Pluto notebook uses @bind for interactivity. When running this notebook outside of Pluto, the following 'mock version' of @bind gives bound variables a default value (instead of an error).
-macro bind(def, element)
-    quote
-        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
-        local el = $(esc(element))
-        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
-        el
-    end
-end
-
```
Comment (cncastillo) [inline examples/6.educational_pluto_notebook/simulation-example.jl:16]: The `@bind Ax ...` object should be present in the  example  (with no solution)

Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/adaa322a7898b1cefc2f2e48124daba339136a7f

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl [hunk 3]
```diff
@@ -162,47 +204,67 @@ $$p_{\Delta w}(w) = \frac{T_2^{'}}{\pi(1+T_2^{'2} w^2)},\quad\text{with }\frac{1
...
 
- - (3.1) Create a copy of the original phantom `obj_t2star = copy(obj)`
- - (3.2) Add a linear distribution of off-resonance to `obj_t2star.Δw .= 2π[-10, 10] rad/s` 
- - (3.3) Plot `obj_t2star` with `plot_phantom_map(obj_t2star, :Δw)` and verify it is correct
+- (3.1) Create a new phantom named `obj_t2star` with spins at the same positions as the original phantom `obj`, each having a linear distribution of the off-resonance parameter. To achieve this, follow these steps:
+   * (3.1.1) Create an empty phantom called `obj_t2star`.
+   * (3.1.2) Iterate over 20 equispaced values starting from -20π up to 20π; these values represent the linear off-resonance distribution.
+   * (3.1.3) Within the loop, create a copy of the original phantom: `obj_aux = copy(obj)` with the off-resonance value of the current iteration.
+   * (3.1.4) Within the loop, update the `obj_t2star` by superimposing it with `obj_aux`.
+   * (3.1.5) Finally, outside the loop, divide the proton density of `obj_t2star` by 20.
```
Comment (cncastillo) [inline examples/6.educational_pluto_notebook/simulation-example-solution.jl:212]:
```suggestion
- (3.1) Create a new phantom named `obj_t2star` with spins at the same positions as the original phantom `obj`, each having a linear distribution of off-resonance. To achieve this, follow these steps:
   * (3.1.1) Create an empty phantom called `obj_t2star`.
   * (3.1.2) Create a linear off-resonance distribution such that the range $2\pi [-10, 10]\,\mathrm{rad/s}` is covered uniformly with $N_{\mathrm{isochromats}} = 20$ (use the function `range(start, stop, length)`).
   * (3.1.3) Iterate over the elements `off` of the linear distribution (`for` loop) and create copies of the original phantom (`obj_aux = copy(obj)`) and set the off-resonance of that copy to `off` with `obj_aux.Δw .= off`.
   * (3.1.4) Update `obj_t2star` by appending the modified copies `obj_aux`.
   * (3.1.5) Finally, outside the loop, divide the proton density `obj_t2star.ρ` by $N_{\mathrm{isochromats}} = 20$ and rename the phantom `obj_t2star.name = "T2 star phantom"`.
```

Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: There is a problem when rendering these changes. This is solved in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/81db429fd3a54aa304d1dd5169f92d879fab01dd.

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl [hunk 4]
```diff
@@ -25,22 +25,22 @@ using KomaMRI, PlutoPlotly, PlutoUI
 
 # ╔═╡ cc66bfed-b61b-4067-8c94-4c54b82a3b42
 md"""
-These are the versions of KomaMRI and their subpackages.
+These are the versions of KomaMRI and its subpackages.
```
Comment (cncastillo) [inline examples/6.educational_pluto_notebook/simulation-example-solution.jl:28]: Revert.

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl [hunk 5]
```diff
@@ -277,73 +299,73 @@ The spin echo experiment has the advantage that the echo signal amplitud it is m
...
 # ╔═╡ 4e1434e1-673f-4206-a271-9edec10ebd6a
 plot_kspace(seq_se)
 
 # ╔═╡ 45952512-aaf1-43d8-a95e-c32bb2633f42
 md"""
- - (4.7) Simulate using `seq_se` and `obj_t2star`
- - (4.8) Compare the signal obtained in (4.6) with the one at (3.5)
- - (4.9) Reconstruct the 1D image
+ - (4.3) Simulate using `seq_se` and `obj_t2star`
+ - (4.4) Compare the signal obtained in (4.6) with the one at (3.5)
```
Comment (cncastillo) [inline examples/6.educational_pluto_notebook/simulation-example-solution.jl:335]: This references are wrong now

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl [hunk 6]
```diff
@@ -277,73 +299,73 @@ The spin echo experiment has the advantage that the echo signal amplitud it is m
...
- - (4.4) A readout gradient of area `2Ax` with an ADC (similar to (2.2)), such that the middle of the gradient and ADC are in $$\mathrm{TE}$$
- - (4.5) Create concatenating these blocks into a sequence called `seq_se`
- - (4.6) Plot `seq_se` and its k-space. Is the k-space the same as `seq_gre` in (2.3)?
+- (4.1) Create a sequence called `seq_se` with the following blocks:
+  * (4.1.1) A 90deg hard RF pulse
+  * (4.1.2) A `Delay` of $$\mathrm{TE}/2$$ with a positive gradient (area `Ax`)
+  * (4.1.3) A 180deg hard RF pulse
+  * (4.1.4) A readout gradient of area `2Ax` with an ADC (similar to (2.2)), such that the middle of the gradient and ADC are in $$\mathrm{TE}$$
+ 
+ - (4.2) Plot `seq_se` and its k-space. Is the k-space the same as `seq_gre` in (2.3)?
```
Comment (cncastillo) [inline examples/6.educational_pluto_notebook/simulation-example-solution.jl:308]: Revert numberings and text changes. The text before this already mentions `seq_se`. I intended to do something like "For this sequence we will need:" It is not necesarry to tell people to create that sequence, is obvious.

File: examples/6.educational_pluto_notebook/simulation-example.jl [hunk 7]
```diff
@@ -245,70 +243,54 @@ The spin echo experiment has the advantage that the echo signal amplitud it is m
...
 
-# ╔═╡ 3b6b91cf-f3ad-40bc-9b3b-8bb5f395537f
-# Run this cell to celebrate!
-html"""
-<script>
-const {default: confetti} = await import("https://cdn.skypack.dev/canvas-confetti@1")
-confetti()
-</script>
-"""
-
```
Comment (cncastillo) [inline examples/6.educational_pluto_notebook/simulation-example.jl:329]: Why did you remove the confetti!? maybe add `activity_finished = false` and `if activity_finished html"""...""" end`.

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 4 above)
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/3fde402ca7a7613045f15f2c77564a4f5491fcfb

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 5 above)
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/3fde402ca7a7613045f15f2c77564a4f5491fcfb

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 6 above)
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/3fde402ca7a7613045f15f2c77564a4f5491fcfb

File: examples/6.educational_pluto_notebook/simulation-example.jl (hunk 7 above)
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/3fde402ca7a7613045f15f2c77564a4f5491fcfb

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl [hunk 8]
```diff
@@ -185,10 +183,9 @@ raw_gre = simulate(obj, seq_gre, sys)
 # ╔═╡ 41d14dec-b852-4316-aefb-c3d08fa43216
 # (2.6) Plot the simulated signal
 begin
-    t_decay = range(10.587, 30.587, 100)
-    trace_decay = scatter(x=t_decay, y=20.0.*exp.(-t_decay ./ 50), name="T2-decay", marker_color="purple")
+    t_adc = range(10.587, 30.587, 100)
```
[the remark this answers is not in the document]
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/3fde402ca7a7613045f15f2c77564a4f5491fcfb

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 6 above)
Comment (cncastillo) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: "For this sequence we will need:"

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 8 above)
[the remark this answers is not in the document]
Comment (cncastillo) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: There is a function ... `get_adc_sampling_times(seq)`

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 4 above)
Comment (cncastillo) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: I want to to be "its"

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 6 above)
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/1b1e5f3d71a7e9f5cd135f14ac9ad9409722a5e9

File: examples/6.educational_pluto_notebook/simulation-example-solution.jl (hunk 8 above)
[the remark this answers is not in the document]
Comment (beorostica) [reply examples/6.educational_pluto_notebook/simulation-example-solution.jl]: Addressed in https://github.com/cncastillo/KomaMRI.jl/pull/253/commits/1b1e5f3d71a7e9f5cd135f14ac9ad9409722a5e9

File: docs/src/pluto-simulation.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# My first MRI simulations: FID, GE, and SE
+# 1D Simulations
```
Comment (cncastillo) [inline docs/src/pluto-simulation.md:1]:
```suggestion
# 1D Simulations: FID, GE, and SE
```
